apps/api/src/scraper/WebScraper/sitemap.ts

```diff
@@ -94,7 +95,17 @@ export async function getLinksFromSitemap(
       }
     }
 
-    const parsed = await parseStringPromise(content);
+    let parsed;
+    try {
+      parsed = await parseSitemapXml(content);
+    } catch (error) {
+      logger.warn("Rust sitemap parsing failed, falling back to JavaScript parser", {
+        method: "getLinksFromSitemap",
+        sitemapUrl,
+        error: error.message,
+      });
+      parsed = await parseStringPromise(content);
+    }
```

Comment:
The Rust version should also replace the part of the logic that comes after this line, as it still consumes a lot of time in JS-land. When using the Rust version, it should send back instructions to JS-land on how to recurse, or what to return.